services/director/tests/unit/conftest.py

```diff
@@ -87,6 +82,23 @@ def configure_registry_access(
     )
 
 
+@pytest.fixture
+def configure_external_registry_access(
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+    external_registry_settings: RegistrySettings | None,
+) -> EnvVarsDict:
+    assert external_registry_settings
```

Comment:
If I don't provide an external `envfile` via the CLI, then `external_registry_settings` should be `None`, correct?
 
Why not simply 
```python

external_envs = external_registry_settings.model_dump(by_alias=True, exclude_none=True) if external_registry_settings else {}

return app_environment | setenvs_from_dict(monkeypatch, {**external_envs})
```